api/v1alpha2/contentlibraryitemimportrequest_types.go

```diff
@@ -0,0 +1,291 @@
+	TransferStatusError TransferStatus = "Error"
+)
+
+// ContentLibraryItemImportRequestSource contains the specification of the
+// source for the import request.
+type ContentLibraryItemImportRequestSource struct {
+	// URL is the endpoint that points to a file that is to be imported as a new
+	// Content Library Item in the target vSphere Content Library. If the target
+	// item type is ContentLibraryItemTypeOvf, the URL should point to an OVF
+	// descriptor file (.ovf), an OVA file (.ova), or an ISO file (.iso).
+	// Otherwise, the SourceValid condition will become false in the status.
+	URL string `json:"url"`
+
+	// +optional
+
+	// PEM encoded SSL Certificate for this endpoint specified by the URL. It is
+	// only used for HTTPS connections.
+	// If set, the remote endpoint's SSL certificate is only accepted if it
+	// matches this certificate, and no other certificate validation is
+	// performed.
+	// If unset, the remote endpoint's SSL certificate must be trusted by
+	// vSphere trusted root CA certificates, otherwise the SSL certification
+	// verification may fail and thus fail the import request.
+	SSLCertificate string `json:"sslCertificate,omitempty"`
+
+	// +optional
+
+	// Checksum contains the checksum algorithm and value calculated for the
+	// file specified in the URL. If omitted, the import request will not verify
+	// the checksum of the file.
+	Checksum *Checksum `json:"checksum,omitempty"`
+}
+
+// Checksum contains the checksum value and algorithm used to calculate that
+// value.
+type Checksum struct {
+	// +optional
+	// +kubebuilder:validation:Enum=SHA256;SHA512
+	// +kubebuilder:default=SHA256
+
+	// Algorithm is the algorithm used to calculate the checksum. Supported
+	// algorithms are "SHA256" and "SHA512". If omitted, "SHA256" will be used
+	// as the default algorithm.
+	Algorithm string `json:"algorithm"`
+
+	// Value is the checksum value calculated by the specified algorithm.
+	Value string `json:"value"`
+}
+
+// ContentLibraryItemImportRequestTargetItem contains the specification of the
+// target content library item for the import request.
+type ContentLibraryItemImportRequestTargetItem struct {
+	// +optional
+
+	// Name is the name of the new content library item that will be created
+	// in vSphere.
+	// If omitted, the content library item will be created with the same name
+	// as the name of the image specified in the spec.source.url in the
+	// specified library.
+	// If an item with the same name already exists in the specified library,
+	// the TargetValid condition will become false in the
+	// status.
+	Name string `json:"name,omitempty"`
+
+	// +optional
+
+	// Description is a description for a library item.
+	Description string `json:"description,omitempty"`
+
+	// +optional
+
+	// Type is the type of the new library item that will be created.
+	//
+	// The valid types depend on the type of underlying library:
+	// - LibraryType=ContentLibrary -- OVF
+	// - LibraryType=Inventory      -- VM
```

Comment:
What would be the import source look like for target item VM type? I presume not the same import from URL.

Reply:
Yeah, this should go away. Mathieu pointed out they don't want to worry about this for 9.1.